Sources/NIO/ChannelHandlers.swift

```diff
@@ -17,6 +17,88 @@
 
 import struct Dispatch.DispatchTime
 
+
+/// `ChannelHandler` which implements some sort of backoff for a `ServerChannel` once `accept` will produce an `IOError`. This is useful as often these errors are recoverable in the sense that we can recover if we slow down the accept rate a bit.
+public final class AcceptBackoffHandler: ChannelDuplexHandler {
+    public typealias InboundIn = Channel
+    public typealias OutboundIn = Channel
+    
+    private var nextReadDeadlineNS: Int?
+    private let backoffProvider: (IOError) -> TimeAmount?
+    private var scheduledRead: Scheduled<Void>?
+    
+    /// Create a new instance
+    ///
+    /// - parameters:
+    ///     - backoffProvider: returns a `TimeAmount` to backoff more `reads` (a.k.a accepts).
+    public init(backoffProvider: @escaping (IOError) -> TimeAmount?) {
```

Comment:
Should this really be implemented in the form of a closure? At the very least shouldn't we provide a default "reasonable estimate" implementation that people can plug in, rather than require them to know how this works?

Reply:
@Lukasa I think the closure thing is nice... maybe we should use a default implementation that use 1 second tho.